### ISSUE TITLE
Using with Axios (Cannot read property 'hasOwnProperty' of undefined)

### ISSUE THREAD

**Issue by ChrisSargent** [captbaritone/raven-for-redux#4]
Hi,

Thanks for this package, I'm looking forward to getting it implemented.

I have a React / Redux universal application that's using Axios to get data from various endpoints. I'm using an Axios interceptor (https://github.com/mzabriskie/axios#interceptors) to catch any network errors like so:

```js
axios.interceptors.response.use((response) => {
    return response;
  }, (error) => {
    Raven.captureException(error)
    return Promise.reject(error);
  });
```

If I run it like this with a default configuration, I get a number of _Cannot read property 'hasOwnProperty' of undefined errors_ relating to Raven._trimBreadcrumbs.

<img width="1005" alt="screen shot 2017-04-10 at 18 45 53" src="https://cloud.githubusercontent.com/assets/12176122/24872554/166f6aa4-1e1e-11e7-94e8-133c19beed6f.png">

On debugging I could see that in the _trimBreadcrumbs function in raven.js, crumb.data was 'undefined' and therefore it couldn't have the hasOwnProperty method. If I add this to the configuration line (it can in fact return any string and it still works okay)

```js
const _ravenBreadcrumbData = (action) => {
  return action.type
}
```

and

```js
createRavenMiddleware(Raven, {breadcrumbDataFromAction: _ravenBreadcrumbData})
```

then it all works okay. So I'm wondering if there's a better way of handling this?

Thanks!

Chris

**Comment by svanzoest** [comment]
This is related to https://github.com/getsentry/raven-js/issues/925

**Comment by captbaritone** [comment]
Looks like [this fix](https://github.com/getsentry/raven-js/pull/929) will be merged in Raven. I have a working fix for `raven-for-redux` but at this point, I think it's best to just wait for the next release of Raven (or downgrade to raven@3.13.1).

You can see the fix on my feature branch here: https://github.com/captbaritone/raven-for-redux/commit/78097c6a4a85bdb2e16c46cad0ebab5c1cc602c2

If anyone would find value in having `raven-for-redux` solve this issue (maybe you can't upgrade/downgrade Raven?) let me know and I can add that commit. In the mean time, I'm going to close this issue.

**Comment by captbaritone** [comment]
In the mean time, I've added a note to the readme: https://github.com/captbaritone/raven-for-redux/commit/137d0c52d895c9d73c861f14054a8a867f663f90

**Comment by svanzoest** [comment]
For now, I just downgraded to raven-js 3.13.1 until getsentry/raven-js#929 gets released.

**Comment by captbaritone** [comment]
Looks like Raven 3.14.1 has been released. Thanks for the fast turn around @benvinegar!

**Comment by ChrisSargent** [comment]
Yip, this is fixed for me after updating to 3.14.1. Thanks!
